### DISCUSSION TITLE
Sharepoint sign in problem

### DISCUSSION THREAD

**Discussion by mar4elom** [pnp/pnpcore#518] [General] [answered]
Hi!

I try to configure access to our company Sharepoint account using my own program with PnP Framework as well as PnP Core SDK - Console Sample, but in both situations I have got error "AADSTS500011: The resource principal named https://login.microsoftonline.com was not found in the tenant named {my-tenant-id}. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant." 
I try to log in with standard user credentials and admin user credentials, but result is the same.

Please help!

**Comment by jansenbe** [comment]
Did you follow this guidance: https://pnp.github.io/pnpcore/using-the-sdk/configuring%20authentication.html#i-want-to-configure-my-own-azure-ad-application-recommended?

**Comment by mar4elom** [comment]
I used this article as guidance: https://github.com/pnp/pnpframework/blob/dev/docs/MigratingFromPnPSitesCore.md
and try to authenticate using demo sample code with my credentials: https://github.com/pnp/pnpcore/tree/dev/samples/Demo.Console.
In both cases my tries are unsuccessful.
Is there any difference between your article and mine?

**Comment by jansenbe** [comment]
Did you update the appsettings.json to reflect your setup (tenant if, app id etc)?

**Reply by mar4elom** [reply]
Yes, sure.

**Reply by jansenbe** [reply]
Did you consent for the needed permissions? Maybe try following the instructions in here https://github.com/pnp/pnpcore/tree/dev/samples/Demo.Console and see if that works for you? Is your tenant a regular tenant (so something like contoso.sharepoint.com?

**Reply by mar4elom** [reply]
I tried to log in with administrator privileges, so I think the problem is something else.
Demo.Console don't work for me (error as in first my message), but probably problem is in my configuration. 
How should look like DemoSiteUrl, DemoSubSiteUrl and CredentialManager?
I should ask our company administrator about what type of tenant we have.

**Reply by jansenbe** [reply]
Here's a sample filled config. For the credential manager option you need to create an entry in your windows credential manager with the name you specify in the config and store the account you're using and it's password. The default option in this sample however is to use the interactive login, if you've plugged in the id of the app you've created and you've correctly set the redirect uri in the app and config this should work. Also cross check the instructions (I've just added additional permissions the app needs) 

```json
{
  "key":  "generic",
  "CustomSettings": {
    "ClientId": "c545f9ce-1c11-440b-812b-0b37118a9e83",
    "TenantId": "d8623c9e-30c7-473a-99de-d907df49a19a",
    "DemoSiteUrl": "https://bertonline.sharepoint.com/sites/prov-1",
    "DemoSubSiteUrl": "https://bertonline.sharepoint.com/sites/prov-1/testsub1",
    "CredentialManager": "bertonline",
    "RedirectUri": "http://localhost"
  },
  "Logging": {
    "LogLevel": {
      "Default": "Information"
    }
  }
}
```

**Comment by mar4elom** [comment]
Hi!

Thank you for your help!
I configured my app according to your example and got another error message
about administration rights.
I contact with our company administrator and he try to log in with admin
credentials today.
Result is different than previously but still unsuccessful, I have got such
error:

[image: image.png]

Which URL should be matched?

trešd., 2021. g. 25. aug., plkst. 15:17 — lietotājs Bert Jansen (<
***@***.***>) rakstīja:

**Reply by jansenbe** [reply]
The screenshot is not visible, please share it again or write down the error you got

**Comment by mar4elom** [comment]
Sorry!
Here is the error message:

"Sorry, but we’re having trouble signing you in.

AADSTS50011: The reply URL specified in the request does not match the
reply URLs configured for the application:
'021bba45-b8e0-41fb-9d96-7493af634bea'."

otrd., 2021. g. 31. aug., plkst. 16:56 — lietotājs Bert Jansen (<
***@***.***>) rakstīja:

**Reply by jansenbe** [reply]
Seems like there's a mismatch in the redirect url specified on the Azure AD app and the one in the settings file. In the configuration I shared earlier there's http://localhost being used, if you've setup something different in your Azure AD app you need to update this configuration.

**Comment by mar4elom** [comment, marked as answer]
Thanks!
The problem was in the Azure AD redirect URL.
But I can log in with global administrator privileges only.
If I try to log in with my regular user it asks to log in as administrator.
What kind of permissions needs regular user to log in through PnP
Framework?

otrd., 2021. g. 31. aug., plkst. 17:16 — lietotājs Bert Jansen (<
***@***.***>) rakstīja:

**Reply by jansenbe** [reply]
The sample in question requires admin approval for the permissions it needs (so your admin needs to perform a grant for all users by clicking on "Grant admin consent")...but that's because the sample code is doing a lot of things...you can also create a Azure AD application with delegated permissions that don't require admin approval, such an app can typically act on the the user's content. If you don't have the needed permissions in your tenant you can start with an app with lesser permissions, ensure the connection works and then step by step add more code

**Reply by mar4elom** [reply]
Finally all configuration problems are solved and authentication works correctly.
Thank you for your help!
